### PR TITLE
UI: <new-list> button that opens up new/edit list settings modal

### PR DESCRIPTION
**Pull request by nsharma123**
https://webarchive.jira.com/browse/WEBDEV-6291

### CODE REVIEW

File: packages/ia-user-list/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+[*.md]
+trim_trailing_whitespace = false
+
+[*.json]
+indent_size = 2
+
+[*.{html,js,md}]
+block_comment_start = /**
+block_comment = *
+block_comment_end = */
```
Comment (iisa) [inline packages/ia-user-list/.editorconfig:29]: q: do we need this?

File: packages/ia-user-list/.github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        node-version: 16.x
+
+    - name: Install dependencies
+      run: yarn install
+
+    - name: Run tests
+      run: yarn run test
+
+    - name: Upload reports
+      run: npx codecov
```
Comment (iisa) [inline packages/ia-user-list/.github/workflows/ci.yml:26]: I'd like for all of these GitHub Actions to actually work if we were to check it in. if not, let's remove them

File: packages/ia-user-list/.gitignore [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+/_site/
+/dist/
+/out-tsc/
+/build/
+ghpages
+dist
+dist-ssr
+*.local
+
+storybook-static
```
Comment (iisa) [inline packages/ia-user-list/.gitignore:32]: you can remove this. we don't use storybook anymore..

File: packages/ia-user-list/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+![Build Status](https://github.com/internetarchive/iaux-typescript-wc-template/actions/workflows/ci.yml/badge.svg) [![codecov](https://codecov.io/gh/internetarchive/iaux-typescript-wc-template/branch/main/graph/badge.svg?token=ZOYRJ2BV9W)](https://codecov.io/gh/internetarchive/iaux-typescript-wc-template)
```
Comment (iisa) [inline packages/ia-user-list/README.md:1]: these don't work here. I don't want anything that doesn't work on merge.. please remove these badges & other references to config that do not work in mono repo.

File: packages/ia-user-list/package.json [hunk 5]
```diff
@@ -0,0 +1,105 @@
+{
```
Comment (iisa) [inline packages/ia-user-list/package.json:1]: note we use yarn. please remove package-lock.json and add yarn.lock

File: packages/ia-user-list/package.json [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  "name": "@internetarchive/ia-userlist-settings",
```
Comment (iisa) [inline packages/ia-user-list/package.json:2]: @bfalling noted we should prefix our components with `<iaux-foo-bar>` please update to `iaux-userlist-settings` for component name

File: packages/ia-user-list/renovate.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": [
```
Comment (iisa) [inline packages/ia-user-list/renovate.json:2]: please remove renovate does not point to this

File: packages/ia-user-list/src/ia-user-list.ts [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+            <button
+              class="ia-button dark"
+              id="cancel"
+              @click=${(event: Event) => {
+                this.emitCloseModalEvent(event);
+              }}
+            >
+              Cancel
+            </button>
+            <button class="ia-button primary">Save</button>
```
Comment (iisa) [inline packages/ia-user-list/src/ia-user-list.ts:133]: noting callback is needed here.

File: packages/ia-user-list/src/models.ts [hunk 9]
```diff
@@ -0,0 +1,6 @@
+export type listDataModel = {
```
[the remark this answers is not in the document]
Comment (iisa) [reply packages/ia-user-list/src/models.ts:1]: you can even move this into the main file with export of course.

File: packages/ia-user-list/test/ia-user-list.test.ts [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+  //     console.log('ssssssssssssssssss');
+  //   });
+
+  //   // Wait for the next frame to ensure the custom event is processed
+  //   // await nextFrame();
+
+  //   // console.log(cancelButton);
+  //   // Check if the custom event was fired
+  //   expect(customEventFired).to.be.true;
+  // });
```
Comment (iisa) [inline packages/ia-user-list/test/ia-user-list.test.ts:116]: noting these are commented out, lmk if we need them?

File: packages/ia-user-list/vite.config.ts [hunk 11]
```diff
@@ -0,0 +1,25 @@
+import { defineConfig } from 'vite';
```
Comment (iisa) [inline packages/ia-user-list/vite.config.ts:1]: we don't do GitHub pages here so we don't need it right now. please remove

File: packages/ia-user-list/index.ts [hunk 12]
```diff
@@ -0,0 +1 @@
+export { IAUserList } from './src/ia-user-list';
```
Comment (iisa) [inline packages/ia-user-list/index.ts:1]: export that type here as well

File: packages/ia-user-list/src/ia-user-list.ts [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+
+import type { listDataModel } from './models';
+import IAButtonStyles from './style/ia-button';
+
+@customElement('ia-user-list')
+export class IAUserList extends LitElement {
+  /**
+   * contains list information
+   *
+   * @type {UserModel}
```
Comment (iisa) [inline packages/ia-user-list/src/ia-user-list.ts:15]: ?
```suggestion
   * @type {UserListDetails}
```

Comment (iisa) [reply packages/ia-user-list/src/ia-user-list.ts:15]: you actually do not need these JSDoc as our types are in-line. please only keep detailed sentence if needed.

File: packages/ia-user-list/src/ia-user-list.ts [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+   * @memberof IAUserList
+   */
+  @property({ type: Object }) listData: listDataModel;
+
+  /**
+   * base api URL for user list service
+   *
+   * @memberof IAUserList
+   */
+  @property({ type: String }) apiURL: string;
```
Comment (iisa) [inline packages/ia-user-list/src/ia-user-list.ts:25]:
```suggestion
  @property({ type: String }) baseHost: string = 'archive.org';
```

We use baseHost everywhere, let's keep using that. + add default if not provided as such.

File: packages/ia-user-list/src/ia-user-list.ts [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+
+  /**
+   * @private
+   * @type {HTMLInputElement}
+   * @memberof IAUserList
```
Comment (iisa) [inline packages/ia-user-list/src/ia-user-list.ts:30]: no need for these jsdoc. ts is our jsdoc 😄

**Review by iisa — CHANGES_REQUESTED** [review]
Neeraj great start.

I was hoping we can absorb the various display of the open modal button here as well.
Given a property boolean, we can show either of the 3 options:
- "+ New List" (for details page drop down)
- "+" (for user list facet column dropdown)
- "✏️ " (for user list header)

This will help streamline modal into Details & OFF Pages. 👍 what do you think?

File: packages/ia-user-list/test/ia-user-list.test.ts (hunk 10 above)
Comment (nsharma123) [reply packages/ia-user-list/test/ia-user-list.test.ts:116]: we need these later

File: packages/ia-user-list/src/ia-user-list.ts (hunk 8 above)
Comment (nsharma123) [reply packages/ia-user-list/src/ia-user-list.ts:133]: Yes, I already use a callback function when you save something

**Comment by nsharma123** [comment]
> Neeraj great start.
> 
> I was hoping we can absorb the various display of the open modal button here as well. Given a property boolean, we can show either of the 3 options:
> 
>     * "+ New List" (for details page drop down)
> 
>     * "+" (for user list facet column dropdown)
> 
>     * "✏️ " (for user list header)
> 
> 
> This will help streamline modal into Details & OFF Pages. 👍 what do you think?

I am not sure how these button will help here except giving just idea like at what places we will show this modal?
This is a standalone component which shows the form in modal.

**Review by nsharma123 — COMMENTED** [review]
I have done the necessary changes here.

File: packages/ia-user-list/package.json (hunk 6 above)
Comment (nsharma123) [reply packages/ia-user-list/package.json]: Currently what I can see on our different repo/component. the repo starts with `/iaux-userlist-settings` but and what we publish is `ia-userlist-settings`. So for now, I am just making consistency here.

**Comment by nsharma123** [comment]
Added three buttons to open Create / Edit form in modal. 
image

**Pull request merged**